Remotion/ObjectBinding/Web.Development.WebTesting/ScreenshotCreation/ScreenshotBocDateTimeValuePickerExtensions.cs

```diff
@@ -172,30 +172,50 @@ private static int GetDayOfTheWeekIndex (DateTime date)
       return (int) date.DayOfWeek - 1;
     }
 
-    private static bool IsVisible (IFluentScreenshotElementWithCovariance<ScreenshotBocDateTimeValuePicker> fluentDatePicker)
+    private static void WaitUntilVisible (IFluentScreenshotElementWithCovariance<ScreenshotBocDateTimeValuePicker> fluentDatePicker, int timeout)
     {
       var dateTimeValue = fluentDatePicker.Target.DateTimeValue;
-      var id = string.Join ("_", dateTimeValue.Scope.Id, "DatePicker");
-      var result = dateTimeValue.Context.RootScope.FindId (id, Options.NoWait);
+      var datePickerID = GetDatePickerID (dateTimeValue);
+      var seleniumDriver = (IWebDriver) dateTimeValue.Context.Browser.Driver.Native;
 
-      return result.Exists (Options.NoWait);
+      var webDriverWait = new WebDriverWait (seleniumDriver, TimeSpan.FromMilliseconds (timeout))

```

Comment:
Couldn't we just pass a TimeSpan to `Open()`?

Reply:
It would be an API change

Reply:
Yeah but it wouldn't be our first, what's the reason why we wouldn't want to do it here?

Reply:
This change would not add any value to the current task.